.github/workflows/push_pull_request_test.yml

```diff
@@ -0,0 +1,16 @@
+name: Automated testing workflow
```

Comment:
Minor suggestion: adding newlines and between sections for readability (similar to the `lint.yml` file).